### DISCUSSION TITLE
Should the package names include the version?

### DISCUSSION THREAD

**Discussion by Teraskull** [microsoft/winget-pkgs#1111] [General]
# Description of the new feature/enhancement

Should the package names include the version?

![image](https://user-images.githubusercontent.com/24798198/82853028-c874ba00-9f04-11ea-971f-789c129e0d97.png)

# Proposed technical implementation details (optional)

I think that the version column is enough.

**Comment by latere-a-latere** [comment]
I don't think it should, simply because there's already a field for that.

**Comment by Teraskull** [comment]
> I don't think it should, simply because there's already a field for that.

Exactly, I will make a PR to remove versions from current package names.

**Comment by StefanKoell** [comment]
I'm the author of [Royal TS](https://github.com/microsoft/winget-pkgs/tree/master/manifests/RoyalApps/RoyalTS) and while I agree that the version should not be in the package name, it would be good to have at least an option to put the "major" version name in the package name somehow. The reason for that is, a new major version may require a new license key and users may not want to upgrade from an older major version to a newer major version using a package manager (automatically using a script) without getting a new license key first. In Chocolatey I maintain two packages at the moment: Royal TS V4 and Royal TS V5.

**Comment by latere-a-latere** [comment]
@StefanKoell I suppose that makes sense, especially since upgrading between two major versions (to take an extreme example, MS Word '97 and MS Word 365) can break a lot of existing workflows.

This should be determined on a case by case basis where the major version has a direct impact on the functionality/licensing/compatibility of the software, because some publishers/developers treat majors differently from others.

**Comment by mimi89999** [comment]
I also think that for some apps it's important to distinguish mayor versions. GNU/Linux distros put it sometimes, especially for programming languages.

**Comment by JiiPee74** [comment]
Yes winget does really need a new feature so you can lock apps to major version.

I provide new example what would benefit from that.

```
PS C:\Users\User> winget upgrade
Name                      Id                        Version     Available       Source
--------------------------------------------------------------------------------------
Razer Synapse             Razer.Synapse             2.21.24.34  3.6.0415.040812 winget

```

I have Synapse 2 installed because Synapse 3 doesn't support my hardware. But winget keeps suggesting upgrade.
Other option would be that Razer would release new version from both and rename them as

> Ryzer Synapse 2
> Ryzer Synapse 3

But I think that would be much more harder to get happen than adding new flag for winget manifest, something like "keep_major" or something like that.

**Comment by denelon** [comment]
We have [Pin a package](https://github.com/microsoft/winget-cli/issues/476) in the CLI backlog.

Many packages (primarily languages) have a version in the package name as a pattern to prevent "major" version upgrades.

Be sure to add your 👍 to that Issue to help raise priority if you feel it is important.

**Comment by jedieaston** [comment]
Once we get schema 1.1 aren't we going to use the PackageVersion to have the "marketing/user-friendly/the one people know" version number anyway? Then we don't need to worry about changing the PackageName from version to version (which isn't supported by wingetcreate at this time, and is a little bit of a pain to automate since you need to do find and replace and might miss something). 

I think it would be much better to just print out the version number next to the package name during `winget install`. That's the only place where this is relevant and the version number isn't shown right now, right? Major version numbers are given different PackageIdentifiers anyway (until version pinning is implemented).
